internal/manager/utils/kongroot.go

```diff
@@ -0,0 +1,181 @@
+package utils
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/avast/retry-go/v4"
+	"github.com/go-logr/logr"
+	"github.com/kong/go-kong/kong"
+	"github.com/samber/lo"
+	"go.uber.org/multierr"
+	"golang.org/x/sync/errgroup"
+)
+
+// ValidateKongRoots checks if all provided kong roots are the same given that we
+// only care about the fact that the following fields are the same:
+// - database setting
+// - kong version.
+func ValidateKongRoots(kongRoots []KongRoot, skipCACerts bool) (string, kong.Version, error) {
+	var (
+		kongVersion kong.Version
+		dbMode      string
+	)
+
+	if err := multierr.Combine(lo.Map(kongRoots, validateKongRootFunc(skipCACerts))...); err != nil {
+		return "", kong.Version{}, fmt.Errorf("failed to validate kong Roots: %w", err)
+	}
+
+	uniqs := lo.UniqBy(kongRoots, getKongRootKeyFunc(skipCACerts))
+	if len(uniqs) > 1 {
+		return "", kong.Version{},
+			fmt.Errorf("there should only be one dbmode:version combination across configured kong instances while there are: %v", uniqs)
+	}
+
+	return dbMode, kongVersion, nil
+}
+
+// KongRoot represents Kong Gateway configuration root.
+type KongRoot map[string]any
```

Comment:
integration tests fails by the following errors:
```
[could not validate Kong admin root](panic: could not validate Kong admin root(s) configuration: failed to validate kong Roots: invalid root configuration, expected a map[string]any got map[string]interface {})
```
Objects are unmarshalled to `map[string]interface{}` from json. They are not the exactly the same type in type assertion.

Reply:
This is fixed now, thanks for pointing this out! 🙇